### ISSUE TITLE
Set up continuous integration with Jenkins for the mono repo

### ISSUE THREAD

**Issue by buckmaxwell** [empirical-org/Empirical-Core#4261]
**This issue will likely be split.**

I am branching off of the the mono repo branch and adding a CI layer with Jenkins.

Goals.
1. Pull requests with passing builds and tests should be automatically merged into develop.
2. Pushes and pull requests to develop should deploy to staging if the build and tests pass
3. Pushes and pull requests to master should deploy to production if the build tests pass
4. Test suite should be run in parallel and optimized for speed in all ways possible.

**Comment by buckmaxwell** [comment]
***TODO**

+ Fix test that occasionally blows up on names that are not strictly title cased (O'Leary, McCrickard, etc.). Even though it doesn't happen every time, it slows builds down to have to rebuild for this dumb reason.
+  Deal with codecov.io parts of build. As of now, they don't seem to do anything.
+ Speed up builds by installing some gems in the docker image
